foreign_cc/cmake.bzl

```diff
@@ -73,6 +62,47 @@ def _create_configure_script(configureParameters):
     flags = get_flags_info(ctx, "<TARGET>")
     no_toolchain_file = ctx.attr.cache_entries.get("CMAKE_TOOLCHAIN_FILE") or not ctx.attr.generate_crosstool_file
 
+    cmake_commands = []
+
+    # If the legacy `make_commands` attribute was not set, use the new
+    # `targets` api for building our target.
+    if not ctx.attr.make_commands:
+        data = ctx.attr.data + getattr(ctx.attr, "tools_deps", [])
+        configuration = "Debug" if is_debug_mode(ctx) else "Release"
+
+        # Generate a list of arguments for cmake's build command
+        build_args = " ".join([
+            ctx.expand_location(arg, data)
+            for arg in ctx.attr.build_args
+        ])
+
+        # Generate commands for all the targets, ensuring there's
+        # always at least 1 call to the default target.
+        for target in ctx.attr.targets or [""]:
+            # Note that even though directory is always passed, the
+            # following arguments can take precedence.
+            cmake_commands.append("{cmake} --build {dir} --config {config} {args} {target}".format(
```

Comment:
How do we ensure that `cmake --build` uses our toolchain for ninja or make? Do we need to make sure we've linked it in the path here under the canonical name? e.g. using a wrapper script to call emscripten wrapped toolchains.
Do cmake presets (https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html) help with this at all?

Reply:
This is solved now by setting [CMAKE_MAKE_PROGRAM](https://cmake.org/cmake/help/latest/variable/CMAKE_MAKE_PROGRAM.html). [presets](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html) could totally help but would only do so in versions of cmake greater than 3.19 so I think it should be something added in the future. For now, `CMAKE_MAKE_PROGRAM` offers all the flexibility we need for the current set of toolchains since it works for the `Unix Makefile` and `Ninja` generators. Though, as stated in the documentation, there are other more nuanced generators that we'd have to solve for in another way if we wanted to have first class support for them

Reply:
Yes, I think its fair enough for the moment to assume that for the windows generators the build tools are installed alongside the compilers.